operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/ResourceController.java

```diff
@@ -3,53 +3,47 @@
-   * is configured to not use a finalizer but the resource does have finalizers, it might be be
-   * "offered" again for deletion several times until the finalizers are all removed.
-   *
-   * @param resource the resource that is marked for deletion
-   * @param context the context with which the operation is executed
-   * @return {@link DeleteControl#DEFAULT_DELETE} - so the finalizer is automatically removed after
-   *     the call. {@link DeleteControl#NO_FINALIZER_REMOVAL} if you don't want to remove the
-   *     finalizer to indicate that the resource should not be deleted after all, in which case the
-   *     controller should restore the resource's state appropriately.
+   * @param resource the {@link CustomResource} instance that is marked for deletion
+   * @param context the context in which the operation is executed
+   * @return either {@link DeleteControl#DEFAULT_DELETE} that indicates that the K8s finalizer on
+   * the resource should automatically be removed as well, or
+   * {@link DeleteControl#NO_FINALIZER_REMOVAL} if you don't want to remove the K8s finalizer, which
+   * will prevent resource deletion by K8s
    */
   default DeleteControl deleteResource(R resource, Context<R> context) {
     return DeleteControl.DEFAULT_DELETE;
   }
 
   /**
-   * The implementation of this operation is required to be idempotent. Always use the UpdateControl
-   * object to make updates on custom resource if possible. Also always use the custom resource
-   * parameter (not the custom resource that might be in the events)
+   * This method is called each time a managed K8s object has been created or updated. The
+   * implementation of this operation must be idempotent, as multiple calls can occur based on
+   * current cluster status.
    *
-   * @param resource the resource that has been created or updated
+   * @param resource the the {@link CustomResource} instance that has been created or updated
    * @param context the context with which the operation is executed
-   * @return The resource is updated in api server if the return value is not {@link
-   *     UpdateControl#noUpdate()}. This the common use cases. However in cases, for example the
-   *     operator is restarted, and we don't want to have an update call to k8s api to be made
-   *     unnecessarily, by returning {@link UpdateControl#noUpdate()} this update can be skipped.
-   *     <b>However we will always call an update if there is no finalizer on object and it's not
-   *     marked for deletion.</b>
+   * @return either an {@link UpdateControl} object, or {@link UpdateControl#noUpdate()} if no
+   * updates need to be sent to the API server.
    */
   UpdateControl<R> createOrUpdateResource(R resource, Context<R> context);
 
   /**
-   * In init typically you might want to register event sources.
+   * Initialization function that registers event sources using a manager object.
```

Comment:
This description really isn't useful. That said, the previous one wasn't all that useful either. Ideally, we'd document the intent of the method, what it needs to be used for, not what it does.